### PR TITLE
Rate limit db writes in known case

### PR DESCRIPTION
**Pull request by bleonard**
## Change description

There are some spots that can create very large SQL writes, so let's break them up.

### Development

- [x] Application changes have been tested appropriately

### Impact

- [x] Code follows company security practices and guidelines
- [x] Security impact of change has been considered
- [x] Performance impact of change has been considered
- [x] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

### Code review

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [ ] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

File: core/src/models/RecordProperty.ts [hunk 1]
```diff
@@ -139,6 +141,30 @@ export class RecordProperty extends CommonModel<RecordProperty> {
...
+    const ids = instances.map((i) => i.id);
+    return this.updateAllinBatchesById(ids, values);
+  }
+
+  static async updateAllinBatchesById(
+    ids: string[],
+    values: { [key: string]: any }
+  ) {
+    const max = config.batchSize.internalWrite;
+    const queue: string[] = Array.from(ids);
```
Comment (evantahler) [inline core/src/models/RecordProperty.ts:158]: Down the rabbit hole I go... https://stackoverflow.com/questions/40548213/array-from-vs-spread-syntax

File: core/src/models/RecordProperty.ts [hunk 2]
```diff
@@ -139,6 +141,30 @@ export class RecordProperty extends CommonModel<RecordProperty> {
...
+  // TODO: I want to figure out the types to move these to CommonModel
+  static async updateAllinBatches(
+    instances: RecordProperty[],
+    values: { [key: string]: any }
+  ) {
+    const ids = instances.map((i) => i.id);
+    return this.updateAllinBatchesById(ids, values);
+  }
+
+  static async updateAllinBatchesById(
```
Comment (evantahler) [inline core/src/models/RecordProperty.ts:153]: Probably could be 1 method

File: core/__tests__/tasks/recordProperty/enqueue/smallerBatchSize.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+process.env.GROUPAROO_RECORDS_BATCH_SIZE = "1";
 process.env.GROUPAROO_INTERNAL_BATCH_SIZE = "1";
```
Comment (evantahler) [inline core/__tests__/tasks/recordProperty/enqueue/smallerBatchSize.ts:2]: Might be interesting to have a test when these aren't the same value

**Review by evantahler — APPROVED** [review]

**Comment by bleonard** [comment]
When setting, to now looks more this this: `GROUPAROO_INTERNAL_BATCH_SIZE=10`
```
2022-01-21T02:05:58.783Z - debug: Executed (16901127-fcd3-4faa-a6bf-105641ef7e6d): UPDATE "recordProperties" SET "startedAt"=$1,"updatedAt"=$2 WHERE "id" IN ('rpr_cf595a0f-9162-44ba-81c3-94d4106810c0', 'rpr_b30e0f8e-be98-436a-9dfd-a2151c70988d', 'rpr_69ef76ca-17b0-4049-884b-5493a032276b', 'rpr_07477ed6-4903-48be-ab70-0b78d5ddf43b', 'rpr_5ff8ee7d-c697-4cc6-a43e-e9daac7c6d27', 'rpr_afcbde8d-b245-418d-99d9-5cf23e0cc15a', 'rpr_f843e326-ce19-4ec3-a54a-59601f139182', 'rpr_f7f63691-a310-4450-a487-bc67efce52fe', 'rpr_44e999f5-ec73-4cae-8c2c-a170a9054a04', 'rpr_376761ca-dd15-4481-9a3d-c6a9f07f64c3') time=1 
2022-01-21T02:05:58.784Z - debug: Executed (16901127-fcd3-4faa-a6bf-105641ef7e6d): UPDATE "recordProperties" SET "startedAt"=$1,"updatedAt"=$2 WHERE "id" IN ('rpr_f6578b7d-2bcd-4163-8f48-c67dec2fd22b', 'rpr_11ffd7a1-2d9e-448e-ba29-69f6a51caa20', 'rpr_a021f019-c644-4a13-b024-dc3100690202', 'rpr_89206fd3-22fe-48b3-98c0-ff2657e781f8', 'rpr_78e268f8-491f-468a-a101-09d5046aa262', 'rpr_45f41e17-9465-4a78-8446-c3fb23a45ce6', 'rpr_4e9067a3-77fc-4fd1-bf3c-7c7f68695e9e', 'rpr_9dd2d188-f1cb-492f-b499-8768b8f3d103', 'rpr_65e36b40-256c-479d-a788-b5a82c582741', 'rpr_3d94fef6-6b71-4617-9600-104e8376e85a') time=1 
2022-01-21T02:05:58.784Z - debug: Executed (16901127-fcd3-4faa-a6bf-105641ef7e6d): UPDATE "recordProperties" SET "startedAt"=$1,"updatedAt"=$2 WHERE "id" IN ('rpr_b99fac94-a764-4896-a8df-563704507d65', 'rpr_6aae1453-2282-4c34-8f3f-031fe2a88f0d', 'rpr_9ae0af34-615b-4f1e-8863-ec72774b16b7', 'rpr_46d8ec5b-1e58-4d62-abcd-fe4a0d1b4c03', 'rpr_8a1e4519-f44f-4253-ab88-5a16a9fd3e10', 'rpr_e3778999-a791-46ab-ae8b-c2b8e513eb10', 'rpr_06ba88fb-2120-43a7-8ea6-ef44b137b7ce', 'rpr_76d1efad-1c14-405f-b3b7-158e7e444bad', 'rpr_1bf189d3-2fa1-4396-b8b2-c8fb1ca1c8ea', 'rpr_5d472dcf-ad32-45c8-bab1-d6b9a10b5227') time=0 
2022-01-21T02:05:58.785Z - debug: Executed (16901127-fcd3-4faa-a6bf-105641ef7e6d): UPDATE "recordProperties" SET "startedAt"=$1,"updatedAt"=$2 WHERE "id" IN ('rpr_bc010f76-6835-4a9f-a625-011a33b45df1', 'rpr_1b910ecb-8cf5-4959-a413-cb4d723ee68d', 'rpr_9d127e3e-3918-4cd6-a111-2f7718676574', 'rpr_c4c55d54-e6ef-4506-8a58-38b51520e3ec', 'rpr_2d119b87-838f-47d0-aa39-2544d5d19910', 'rpr_2d43b3a2-a12e-405f-9d14-d371950fa0d3', 'rpr_16d63f4f-a7f0-48c1-8415-37caba9c6df7', 'rpr_6c8fce87-b579-4601-841a-cdd39d9a2a1f', 'rpr_dd9ea7df-0
```

File: core/src/models/RecordProperty.ts (hunk 2 above)
Comment (bleonard) [reply core/src/models/RecordProperty.ts:153]: True. I was modeling what I hoped to get into `CommonModel` if I wasn't thwarted by TS.

**Pull request merged**